### PR TITLE
trcplugin-search controls v2

### PR DESCRIPTION
**Pull request by dinohamzic**
This PR has the following changes:
- Creation of Button, Panel, SelectInput, HorizontalList components
- Update existing Questions control to consume newly created components

This is an updated screenshot of `trcplugin-search` (no custom CSS, everything is component based):
localhost_3000_index html

### CODE REVIEW

File: src/Button.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import styled from '@emotion/styled';
+
+interface IProps {
+  secondary?: boolean
+}
+
+export const Button = styled.button<IProps>`
```
Comment (VoterScience) [inline src/Button.tsx:7]: I wonder if we end up with 2 layers to this library: things are depend on TRC (<sheetContainer>, etc) and things that are truly generic (like Button, HorizontalList). Maybe we use the generic ones in our other projects (where we don't have the TRC APIs)

Comment (dinohamzic) [reply src/Button.tsx:7]: I see what you are saying. We could play it out and see what happens, and if the need arises we can split these type of components in a more generic library.

Comment (VoterScience) [reply src/Button.tsx:7]: What if for starters, we kept in same library (as that definitely simplifies the mechanics) but split them by directory? Had a "common" directory for the generic controls.  If that doesn't work, ok.

**Comment by dinohamzic** [comment]
> What if for starters, we kept in same library (as that definitely simplifies the mechanics) but split them by directory? Had a "common" directory for the generic controls. If that doesn't work, ok.

@VoterScience that's what I wanted to propose as well, since when the size of the library inevitably increases with new components, we would need a better folder structure either way. I can do that in a separate PR. 👍

**Pull request merged**